src/client/client.go

```diff
@@ -245,23 +248,43 @@ func (c *APIClient) connect() error {
 		return err
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
+	c._ctx, c.cancel = context.WithCancel(context.Background())
 	c.PfsAPIClient = pfs.NewAPIClient(clientConn)
 	c.PpsAPIClient = pps.NewAPIClient(clientConn)
 	c.ObjectAPIClient = pfs.NewObjectAPIClient(clientConn)
+	c.AuthAPIClient = auth.NewAPIClient(clientConn)
 	c.clientConn = clientConn
 	c.healthClient = health.NewHealthClient(clientConn)
-	c._ctx = ctx
-	c.cancel = cancel
 	return nil
 }
 
-func (c *APIClient) addMetadata(ctx context.Context) context.Context {
+// RequestCtx is a convenience function that returns a sensible default
+// context for top-level Pachyderm RPCs sent by the user (though it does not
+// assume a deadline, in case it's used by the command-line client.
+// Note that the client may cancel in-flight requests using this context if
+// it believes the server has become unavailable.
```

Comment:
Do these changes still make sense given that we are using the Auth API directly?